platform/include/retry_utils.h

```diff
@@ -35,40 +35,56 @@
 #if defined( __cplusplus ) || ( ( defined( __STDC_VERSION__ ) ) && ( __STDC_VERSION__ >= 199901L ) )
     #include <stdbool.h>
 #elif !defined( bool )
-    #define bool     signed char
```

Comment:
We might want to change the `!defined( bool )` to `!defined( bool ) && !defined( false ) && !defined( true )` to match mqtt, but that's unrelated to this PR

Reply:
Oh let's change it now then.